### PR TITLE
Af packet fixes 6

### PR DESCRIPTION
**Pull request by regit**
Update bypass code to deal with flow engine update.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
- https://redmine.openinfosecfoundation.org/issues/3861
- https://redmine.openinfosecfoundation.org/issues/3902

Describe changes:
- Fix 2 description texts
- Update code to new API

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/539
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/315

### CODE REVIEW

File: src/flow-hash.c [hunk 1]
```diff
@@ -849,20 +849,25 @@ static inline int FlowCompareKey(Flow *f, FlowKey *key)
...
 
     if (f != NULL) {
         return f;
     }
-    /* TODO use spare pool */
-    /* now see if we can alloc a new flow */
-    f = FlowAlloc();
+
+    /* get a flow from the spare queue */
+    f = FlowQueuePrivateGetFromTop(&fls->spare_queue);
```
Comment (victorjulien) [inline src/flow-hash.c:862]: this is still missing a step. The pool remains empty. The spare pool is now a per thread construct, where it gets a 'budget' from the master pool. See https://github.com/OISF/suricata/blob/master/src/flow-hash.c#L551 in the normal lookup path.

Comment (regit) [reply src/flow-hash.c:862]: OK, there is a problem with FlowSpareSync: What am I suppose to use for packet, should I created a pseudo packet ? or should I cook a derivative of FlowSpareSync that don't need a Packet ?

Comment (regit) [reply src/flow-hash.c:862]: By looking at FlowSpareSync I could maybe just patch it to support a NULL value for Packet as it is only used for emergency mode. This way, we would have a regular call here.

Comment (regit) [reply src/flow-hash.c:862]: Hum I could just call it with emerg set to false and this should pass.

Comment (victorjulien) [reply src/flow-hash.c:862]: Or create your own version of the FlowSpareSync func.

**Pull request closed without merging**